### PR TITLE
native: more low-priority fixes

### PR DESCRIPTION
**Pull request by jamesacklin**
- Eliminates the search icon in non-Chat channels
- Proper dark mode styling for the SearchBar component
- Adds a dummy shim into the Activity view until we're ready
- Changes the opacity of swiped SwipableChatListItem (more refinements to come, I'm sure)
- Eliminates the gray box behind transparent PNGs in ChatList
- Eliminates contact colors rendering behind transparent PNG avatars
- Adds CalmEngine support to ChatList (reviewed in #3600)
- Adds a getBackgroundColor helper to handle the above

### CODE REVIEW

**Review by patosullivan — APPROVED** [review]
Looks good

**Comment by jamesacklin** [comment]
Review #3600 first, please -- that's branched from this PR. (My bad.)

File: packages/ui/src/components/SearchBar.tsx [hunk 1]
```diff
@@ -63,9 +63,9 @@ export function SearchBar({
             justifyContent="center"
             alignItems="center"
             size="$xl"
-            backgroundColor="$color.gray500"
+            backgroundColor="$secondaryText"
```
Comment (dnbrwstr) [inline packages/ui/src/components/SearchBar.tsx:66]: 🙏

File: packages/ui/src/components/SwipableChatListItem.tsx [hunk 2]
```diff
@@ -25,8 +27,34 @@ export function SwipableChatRow(
...
 
+  const opacity = swipeAnim.interpolate({
+    inputRange: [0, 1],
+    outputRange: [1, 0.5],
+  });
+
   return (
     <Swipeable
+      onSwipeableWillOpen={() =>
+        Animated.timing(swipeAnim, {
```
Comment (dnbrwstr) [inline packages/ui/src/components/SwipableChatListItem.tsx:38]: Should prob memoize these guys -- rerenders here are pretty perf sensitive.

File: packages/ui/src/components/GroupListItem/GroupListItemContent.tsx [hunk 3]
```diff
@@ -27,8 +30,18 @@ export default function GroupListItemContent({
     >
       <ListItem.Icon
         fallbackText={model.title?.[0]}
-        backgroundColor={model.iconImageColor ?? undefined}
-        imageUrl={model.iconImage ?? undefined}
+        backgroundColor={
+          disableAvatars
```
Comment (dnbrwstr) [inline packages/ui/src/components/GroupListItem/GroupListItemContent.tsx:34]: Are we applying this logic anywhere else? At some point we probably want this broken out as a helper, but doesn't have to be now.

Comment (dnbrwstr) [reply packages/ui/src/components/GroupListItem/GroupListItemContent.tsx:34]: Same for the image url below.

File: packages/ui/src/components/Avatar.tsx [hunk 4]
```diff
@@ -26,9 +26,13 @@ export function Avatar({
       size={size}
       {...props}
       // @ts-expect-error custom color
-      backgroundColor={colors.backgroundColor}
+      backgroundColor={
+        contact?.avatarImage && !disableAvatars
```
Comment (dnbrwstr) [inline packages/ui/src/components/Avatar.tsx:30]: Maybe helper (see comment below)

File: apps/tlon-mobile/src/hooks/useCalmSettings.ts
Comment (dnbrwstr) [inline apps/tlon-mobile/src/hooks/useCalmSettings.ts:1]: 🙏

**Review by dnbrwstr — APPROVED** [review]
Looking good, some minor nits.

File: packages/ui/src/components/GroupListItem/GroupListItemContent.tsx (hunk 3 above)
Comment (jamesacklin) [reply packages/ui/src/components/GroupListItem/GroupListItemContent.tsx]: Cool, at least abstracted backgroundColor out to handle both this and Avatar. PTAL!

**Review by dnbrwstr — APPROVED** [review]
Beautiful, lgtm!

**Pull request merged**